### PR TITLE
Cmake and conan revamp

### PR DESCRIPTION
**Pull request by madebr**
Hello!

As promised in https://github.com/conan-io/conan-center-index/pull/7317/files#r710556300, here is a pr with improvements (imho) to your build system + tests.

First, I removed all references to conan from the main `CMakeLists.txt`.
It was currently also possible to build without conan as the `conan_paths` include was OPTIONAL,
but it's better (imho) to don't include it if not needed.
The conan build process is completely controlled by conan now.
You can create a `edyn` package in the conan cache using `conan create .`.
For local builds, you can do:
```
mkdir build
cd build
conan install ..
conan build ..
```

~I've also created a `test_package/conanfile.py` that re-uses your tests + examples.
(`test/edyn/shapes/test_paged_trimesh.cpp` does not build on gcc11)~
I reverted this because it felt strange to pull the project in two. Instead, I added some options to build the tests/examples, enable the assertions or sanitizers.

Second, the cmake script now allows to install `edyn` like you would install any other project.
It will also install an `Edyn-config.cmake` file, such that users of your library can simply do `find_package(Edyn REQUIRED)`.

Third, I wrote a little github workflow yml to build a static edyn for all 3 major platforms (linux, macos and windows).
It looks like [edyn fails to build on gcc9@Linux](https://github.com/madebr/edyn/runs/3640771626?check_suite_focus=true).
Perhaps edyn requires gcc 10 or higher?

Fourth, I added a `.editorconfig` file to autoconfigure IDE's.
It's IDE-independent and supported by a few. See https://editorconfig.org/.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -152,10 +179,47 @@ if(EDYN_BUILD_EXAMPLES)
...
+    install(
+        DIRECTORY include/edyn/ src/edyn/ ${PROJECT_BINARY_DIR}/include/edyn/
+        DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/edyn"
+        FILES_MATCHING
+            PATTERN "*.h"
+            PATTERN "*.hpp"
+    )
+    install(
+        EXPORT EdynTargets
+        DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/Edyn"
```
Comment (madebr) [inline CMakeLists.txt:204]:
```suggestion
        DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}"
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -152,10 +179,47 @@ if(EDYN_BUILD_EXAMPLES)
...
+        NAMESPACE "Edyn::"
+        FILE "Edyn-targets.cmake"
+    )
+    install(
+        FILES
+            cmake/Edyn-config.cmake
+        DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/Edyn"
```
Comment (madebr) [inline CMakeLists.txt:211]:
```suggestion
        DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}"
```

File: CMakeLists.txt [hunk 3]
```diff
@@ -101,45 +110,63 @@ set(Edyn_SOURCES
...
-target_sources(Edyn PRIVATE ${Edyn_SOURCES})
-
-configure_file(${Edyn_SOURCE_DIR}/cmake/in/build_settings.h.in ${Edyn_BINARY_DIR}/include/edyn/build_settings.h @ONLY)
-install(FILES ${Edyn_BINARY_DIR}/include/build_settings.h DESTINATION include/edyn)
+if(WIN32)
+    target_sources(Edyn PRIVATE
+        src/edyn/time/windows/time.cpp
+    )
+    target_link_libraries(Edyn
+        PRIVATE
```
Comment (madebr) [inline CMakeLists.txt:134]:
```suggestion
        PUBLIC
```

File: CMakeLists.txt [hunk 4]
```diff
@@ -101,45 +110,63 @@ set(Edyn_SOURCES
...
     )
 endif()
 
-if(WIN32)
-    set(Edyn_SOURCES
-        ${Edyn_SOURCES}
-	src/edyn/time/windows/time.cpp
+if(UNIX AND NOT APPLE)
+    target_link_libraries(Edyn
+        PRIVATE
```
Comment (madebr) [inline CMakeLists.txt:123]:
```suggestion
        PUBLIC
```

**Comment by xissburg** [comment]
Thank you so much for this. There's no way I could've done most of this myself.

I wonder if the build failure is due to it not compiling with C++17? E.g. `/constraint.hpp:36:10: error: ‘apply’ is not a member of ‘std’`

**Comment by madebr** [comment]
See #58 
Check the action logs for the errors.

**Pull request closed without merging**